packages/gatsby-plugin-theme-ui/src/provider.js

```diff
@@ -20,8 +23,12 @@ const Root = ({ children }) => {
   const fullTheme = merge(themeWithPrism, localTheme)
 
   return (
-    <ThemeProvider theme={fullTheme} components={components}>
-      {children}
+    <ThemeProvider theme={fullTheme}>
+      <MDXProvider
+        components={useThemedStylesWithMdx(useMDXComponents(components))}
```

Comment:
I'm not sure if it has an impact on rendering performance, but is it preferable to pull this out into a variable?